test/test_mocking_multi_invocations.sh

```diff
@@ -32,6 +32,24 @@ test_mocking_a_command_to_return_a_value_and_exit_failure() {
     assert "${result}" equals "hello"
 }
 
+test_mocking_a_command_to_return_a_value_on_stderr_and_exit_success() {
+    mock some-command --and echo-stderr "hello" --and exit-code 0
+
+    result=$(some-command 2>&1 1>/dev/null)
+    assert ${?} succeeded
+
+    assert "${result}" equals "hello"
+}
+
+test_mocking_a_command_to_return_a_value_on_stderr_and_exit_failure() {
+    mock some-command --and echo-stderr "hello" --and exit-code 1
+
+    result=$(some-command 2>&1 1>/dev/null)
+    assert ${?} failed
+
+    assert "${result}" equals "hello"
+}
+
 test_mocking_a_command_to_return_content_from_missing_file() {
     mock some-command --and cat /missing/file
     result=$(some-command)
```

Comment:
fix that also, thank you come again

Reply:
It doesn't look like he came back again. Is he ever coming back?

Reply:
Just havent had the time to remove unused variable assignment :) Wanna do it Waj?

Reply:
(been almost a year lol)